### DISCUSSION TITLE
TFT_BACKLIGHT_ON not used in ST7789 and settings seem opposite those I'm testing with and are working

### DISCUSSION THREAD

**Discussion by ljunquera** [Bodmer/TFT_eSPI#2371] [Alerts]
I am trying to work with the ST7789 but am wondering why the `TFT_BACKLIGHT_ON` setting is not used. I see in the driver's [init](https://github.com/Bodmer/TFT_eSPI/blob/bd33fc3ff904c682fccea52c288f40ae207d65f3/TFT_Drivers/ST7789_Init.h) file I see this in two places:

```
#ifdef TFT_BL
  // Turn on the back-light LED
  digitalWrite(TFT_BL, HIGH);
  pinMode(TFT_BL, OUTPUT);
#endif
```

In my code, just testing the backlight, this makes it turn on:
`digitalWrite(TFT_BL, LOW);`
and this makes it turn off:
`digitalWrite(TFT_BL, HIGH);`

A lot of the other drivers use it. I also see this code in the [TFT_eSPI.cpp](https://github.com/Bodmer/TFT_eSPI/blob/c6d600b4a1ef247e274ac61fa2a4cc13fe726c1d/TFT_eSPI.cpp):

```
#if defined (TFT_BL) && defined (TFT_BACKLIGHT_ON)
  pinMode(TFT_BL, OUTPUT);
  digitalWrite(TFT_BL, TFT_BACKLIGHT_ON);
#else
  #if defined (TFT_BL) && defined (M5STACK)
    // Turn on the back-light LED
    pinMode(TFT_BL, OUTPUT);
    digitalWrite(TFT_BL, HIGH);
  #endif
#endif
```
Does this override the driver settings?

**Comment by Bodmer** [comment]
Different displays have different arrangments for the backlight.  Some displays have a transistor that can be used to switch the backlight on and off. The majority of displays do not have a backlight that is controlled via the display data/command interface. Some displays require power to be applied to the backlight signal. (e.g 3.3V or 5V).

**Comment by ljunquera** [comment]
If that's the case shouldn't the ST7789_init.h driver have a TFT_BACKLIGHT_ON setting vs hard coding HIGH:

```
#ifdef TFT_BL
  // Turn on the back-light LED
  digitalWrite(TFT_BL, HIGH);
  pinMode(TFT_BL, OUTPUT);
#endif
```

should be this:

```
#ifdef TFT_BL
  // Turn on the back-light LED
  digitalWrite(TFT_BL, TFT_BACKLIGHT_ON);
  pinMode(TFT_BL, OUTPUT);
#endif
```

**Comment by ljunquera** [comment]
I'm curious why this would not be an issue given there is a fix requested to the `ST7789_Init.h` to support different displays which have different arrangements for the backlight driver using the TFT_BACKLIGHT_ON, as opposed to a discussion topic?

**Comment by Bodmer** [comment]
There is legacy code in the ST7789_init.h driver. You can delete it if you wish.

**Reply by ljunquera** [reply]
So the two places the ST7789_init.h driver where this code exists should be deleted:

```
#ifdef TFT_BL
  // Turn on the back-light LED
  digitalWrite(TFT_BL, HIGH);
  pinMode(TFT_BL, OUTPUT);
#endif
```
